lib/wasi/src/state/builder.rs

```diff
@@ -56,15 +57,23 @@ impl std::fmt::Debug for WasiStateBuilder {
 }
 
 /// Error type returned when bad data is given to [`WasiStateBuilder`].
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Error, Debug, PartialEq, Eq)]
```

Comment:
Note, this is needed because `anyhow` (used in the wasi tests) needs to receive an Rust error, and it throw me a compiler error when trying to map our errors to anyhow.

So it's actually something that makes our API better as well!
I prefered to do it in one stand, rather than in a PR apart because otherwise I will forget.